Makefile

```diff
@@ -339,7 +339,7 @@ mango-test: devclean all
 		--admin=adm:pass \
 		--no-eval "\
 COUCH_USER=adm COUCH_PASS=pass \
-src/mango/.venv/bin/nose2 -s src/mango/test -c src/mango/unittest.cfg $(MANGO_TEST_OPTS)"
+src/mango/.venv/bin/nose2 -F -s src/mango/test -c src/mango/unittest.cfg $(MANGO_TEST_OPTS)"
```

Comment:
What does `-F` buy for us?  The Mango tests are quick to run, and sometimes perhaps it is good to see if there are other errors besides the one that failed the whole run.

Reply:
While working on this, I even changed `unittest.cfg` to run `16-index-selectors-test.py` only. Because I keep getting 500 errors due to this test file. Different combinations of test cases will result in the same failure message but fail on different test cases. 0.85 sec vs 50 sec to run the entire test suite.

`-F, --failfast` allows us to identify the issue quickly, speed up the feedback loop, and make debugging easier.

Reply:
> 0.85 sec vs 50 sec to run the entire test suite.

Sure, I know that running the Mango test suite in the past required around a minute to run.  In my opinion, that is acceptable, since it is still does not take tens of minutes or many hours.

> Different combinations of test cases will result in the same failure message but fail on different test cases.

I am not sure if I got you on this.  If the "fail fast" mode stops at the first failing test, how can it result in the failure message if there are multiple failures?

> `-F, --failfast` allows us to identify the issue quickly, speed up the feedback loop, and make debugging easier.

Note that you could tell the names of the modules or specific test cases on the command line when invoking `nose2`.  This is also exposed in the general `Makefile` of the repository, for the `mango-test` target as the `MANGO_TEST_OPTS` variable.

With the help of this, one could easily limit the current run to the tests that one has to run for keeping the feedback loop short.  With the regards to finding the tests to run, it is okay to run the whole suite once and take the cost of the full run.

Reply:
> Same error message, but bailed on different test cases.

The error message is as follows:
```bash
Traceback (most recent call last):
  File "/Users/jiahuili/src/a/src/mango/test/16-index-selectors-test.py", line 214, in test_uses_partial_index_for_query_selector
    docs = self.db.find(selector, use_index="Selected", fields=["_id", "location"])
  File "/Users/jiahuili/src/a/src/mango/test/mango.py", line 317, in find
    r.raise_for_status()
    ~~~~~~~~~~~~~~~~~~^^
  File "/Users/jiahuili/src/a/src/mango/.venv/lib/python3.13/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://127.0.0.1:15984/mango_test_480b738790f748d9b5f3c5c17202d501/_find
```

Sometimes it fails on `test_uses_partial_index_for_query_selector` and sometimes it fails on `test_uses_partial_index_with_different_selector`. Comment out some test cases and you'll get different test results.

We can remove the `-F` if it is not needed: https://github.com/apache/couchdb/pull/5348/files

Reply:
> Comment out some test cases and you'll get different test results.

That is what one would not want to do.  Commenting test cases in and out is what makes the feedback loop slower.

> We can remove the `-F` if it is not needed: https://github.com/apache/couchdb/pull/5348/files?w=1

I have not had the time to try this new approach, maybe it works.